### DISCUSSION TITLE
graphql-dgs-example-java-shared and graphql-dgs-example-java

### DISCUSSION THREAD

**Discussion by sbilello** [Netflix/dgs-framework#516] [General]
Hi everyone!

Would it make sense to keep the example self-contained without referencing the external module  `graphql-dgs-example-java-shared`?

**Comment by paulbakker** [comment]
The reason for the split is so that we can run the same example code with
different stacks: WebMVC and WebFlux, and in the near future
graphql-spring. It makes it a little harder to navigate the examples, but
it avoids the duplication we would otherwise get.

**Reply by sbilello** [reply]
The reason is pretty obvious.
I expect a developer that would grab just one of the modules to get familiar with the framework and would discover that it is missing some parts.
Readability and Isolation vs Code Duplication? :)

**Reply by berngp** [reply]
I think there is a point on calling them examples thought. Might be better to rename and point users to the following in the README file.

* https://github.com/Netflix/dgs-federation-example
* https://github.com/Netflix/dgs-examples-webflux
* https://github.com/Netflix/dgs-examples-java16
* https://github.com/Netflix/dgs-examples-webflux
* https://github.com/Netflix/dgs-examples-java
* https://github.com/Netflix/dgs-examples-kotlin

**Reply by paulbakker** [reply]
Yeah good point - the "examples" in the main repo aren't really user-focussed, but I can see how that's not obvious.
